src/main/java/com/kenshoo/swagger/validator/MimeValidator.java

```diff
@@ -0,0 +1,275 @@
+                );
+            }
+        }
+    }
+
+    /**
+     * Validates that root defined mime types are valid
+     *
+     * @param producesClassAnnotation
+     * @param consumesClassAnnotation
+     */
+    private void validateRootMimes(Produces producesClassAnnotation, Consumes consumesClassAnnotation) {
+        validateRootMime("produces", producesClassAnnotation);
+        validateRootMime("consumes", consumesClassAnnotation);
+    }
+
+    /**
+     * Validates specific mime that defined in
+     * "Definitions" area of the yaml (i.e root mime)
+     *
+     * @param key the mime name
+     * @param mimeAnnotation actual annotation representing this mime
+     */
+    private void validateRootMime(String key, Annotation mimeAnnotation) {
+        List<String> rootMimes = (List<String>)genericMetadata.get(key);
+
+        if (mimeAnnotation != null) {
+            String mimeAnnotationValue = getAnnotationValue(mimeAnnotation);
+            if (rootMimes != null && rootMimes.size() >= 1 &&
+                    !Strings.isNullOrEmpty(mimeAnnotationValue) &&
+                    !mimeAnnotationValue.equalsIgnoreCase(rootMimes.get(0))) {
+                messageEventHandler.handleError(
+                        "{0} root mime {1} does not match {2} defined in yaml.",
+                        key.toUpperCase(),
+                        mimeAnnotationValue,
+                        rootMimes.get(0)
+                );
+            }
+        }
+    }
+
+    /**
+     * Extracts annotation value.
+     *
+     * @param annotation
+     * @return
+     */
+    private String getAnnotationValue(Annotation annotation) {
+        if (annotation != null) {
+            String value = annotation.toString();
+            return value.substring(
+                    value.indexOf("[", 0) + 1,
+                    value.lastIndexOf("]")
+            );
+        }
+        return null;
+    }
+
+    /**
+     * Extracts actual mime from doc string
+     *
+     * @param producesDocumentation
+     * @return MimeType
+     */
+    private String getMimeType(String producesDocumentation) {
+        return producesDocumentation != null &&
+                producesDocumentation.startsWith("[") && producesDocumentation.endsWith("]") ?
+                producesDocumentation.substring(1, producesDocumentation.length()-1)
+                :
+                producesDocumentation;
+    }
+
+    /**
+     * Fetches method corresponding to given jax-rs annotation.
+     *
+     * @param cls class to look at
+     * @param httpMethod jax-rs annotation
+     *
+     * @return Method
+     */
+    private Method getJaxRsMethod(Class<?> cls, String httpMethod) {
+        Class<? extends Annotation> annotationClass = resolveJaxRsAnnotation(httpMethod);
+        for (Method m : cls.getMethods()) {
+            Object jaxrsMethodAnnotation = m.getAnnotation(annotationClass);
+            Object pathAnnotation = m.getAnnotation(Path.class);
```

Comment:
here i'm looking for method corresponding to yaml definition (i.e get/post/etc.), 
thus i have exact http method to look for.